src/file.cpp

```diff
@@ -91,16 +97,67 @@ file file::copy(const fs::path& path, std::string_view label,
 }
 
 std::string file::read() const {
-  std::ifstream stream = input_stream();
-  return std::string(std::istreambuf_iterator<char>(stream), {});
+#ifdef _WIN32
+  int mode = _O_RDONLY | (binary ? _O_BINARY : _O_TEXT);
+  std::intptr_t osfhandle = reinterpret_cast<std::intptr_t>(native_handle());
+
+  int handle = _open_osfhandle(osfhandle, mode);
+#else
+  native_handle_type handle = native_handle();
+  lseek(handle, 0, SEEK_SET);
+#endif
+
+  std::string content;
+  content.resize(file_size(path()));
```

Comment:
Although we can read exactly `file_size` bytes on POSIX and no more than `file_size` on Windows, I don't like this solution

We should properly check if we are still in bounds or use something like string stream

Reply:
File size may change during reading

While this would be undefined behavior for `tmp`, it would be nice to not have a crash here

Reply:
Also, file_size can throw an exception without our label